### PR TITLE
genai/test: add example for Tool / FunctionCall

### PR DESCRIPTION
**Pull request by eliben**

### CODE REVIEW

File: genai/example_test.go [hunk 1]
```diff
@@ -258,6 +259,102 @@ func ExampleClient_ListModels() {
...
+func ExampleTool() {
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("GEMINI_API_KEY")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	// To use functions / tools, we have to first define a schema that describes
+	// the function to the model. The schema is similar to OpenAPI 3.0
```
Comment (jba) [inline genai/example_test.go:271]: nit: period at end of sentence

File: genai/example_test.go [hunk 2]
```diff
@@ -258,6 +259,102 @@ func ExampleClient_ListModels() {
...
+	weatherTool := &genai.Tool{
+		FunctionDeclarations: []*genai.FunctionDeclaration{{
+			Name:        "CurrentWeather",
+			Description: "Get the current weather in a given location",
+			Parameters:  schema,
+		}},
+	}
+
+	model := client.GenerativeModel("gemini-1.0-pro")
+	model.SetTemperature(0)
```
Comment (jba) [inline genai/example_test.go:299]: I did this in the test for better reproducibility but it's a distraction in this example.

File: genai/example_test.go [hunk 3]
```diff
@@ -258,6 +259,102 @@ func ExampleClient_ListModels() {
...
+		log.Fatalf("expected CurrentWeather: %v", funcall.Name)
+	}
+
+	// Expect the model to pass a proper string "location" argument to the tool.
+	locArg, ok := funcall.Args["location"].(string)
+	if !ok {
+		log.Fatalf("expected string: %v", funcall.Args["location"])
+	}
+
+	if strings.Contains(locArg, "New York") {
```
Comment (jba) [inline genai/example_test.go:341]: Call `currentWeather` here.

**Review by jba — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: genai/example_test.go (hunk 1 above)
Comment (eliben) [reply genai/example_test.go]: Done

File: genai/example_test.go (hunk 2 above)
Comment (eliben) [reply genai/example_test.go]: Done

File: genai/example_test.go (hunk 3 above)
Comment (eliben) [reply genai/example_test.go:340]: Not sure what you mean -- I don't have a `currentWeather` function? Do you mean I should create one and call it here?

Comment (jba) [reply genai/example_test.go:340]: Looks like I forgot to save a comment.
I think at the top of the function you should define
```
currentWeather := func(city string) string {
    switch city {
    case "New York": return "cold"
    case "San Francisco": return "warm"
    default: return "unknown"
}
```

Comment (eliben) [reply genai/example_test.go]: Done!

**Review by jba — APPROVED** [review]

**Pull request merged**